miner/scroll_worker.go

```diff
@@ -592,18 +592,20 @@ func (w *worker) startNewPipeline(timestamp int64) {
 }
 
 func (w *worker) handlePipelineResult(res *pipeline.Result) error {
+	startingHeader := w.currentPipeline.Header
+	w.currentPipeline.Release()
```

Comment:
is it better to add a nil check of `res` after this? or `nil` res never happen?

Reply:
but I think the result would be the same (panic of the node).

Reply:
some more context here https://github.com/scroll-tech/go-ethereum/pull/788#discussion_r1621935061

and our discussion with Peter

```
all sites that call TryPushTxnscheck that the returned result is not-nil before calling handle

like this
				if result := w.currentPipeline.TryPushTxns(txset, w.onTxFailingInPipeline); result != nil {
					w.handlePipelineResult(result)
				}

only case that does this without a nil check is;

		case result := <-pipelineResultCh():
			w.handlePipelineResult(result)
```

Reply:
got it.